java/arcs/android/storage/database/DatabaseImpl.kt

```diff
@@ -0,0 +1,228 @@
+                    id INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
+                    name TEXT NOT NULL UNIQUE,
+                    is_primitive INTEGER NOT NULL DEFAULT 0
+                )
+
+                CREATE INDEX type_name_index ON types (name, id)
+
+                CREATE TABLE storage_keys (
+                    id INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
+                    storage_key TEXT UNIQUE NOT NULL
+                )
+
+                CREATE INDEX storage_key_index ON storage_keys (storage_key, id)
+
+                CREATE TABLE entities (
+                    storage_key_id INTEGER NOT NULL PRIMARY KEY,
+                    type_id INTEGER NOT NULL
+                )
+
+                CREATE TABLE collections (
+                    storage_key_id INTEGER NOT NULL PRIMARY KEY,
+                    type_id INTEGER NOT NULL
+                )
+
+                CREATE TABLE collection_entries (
+                    collection_storage_key_id INTEGER NOT NULL,
+                    entity_storage_key_id INTEGER NOT NULL
+                )
+
+                CREATE INDEX
+                    collection_entries_collection_storage_key_index
+                ON collection_entries (collection_storage_key_id)
+
+                CREATE TABLE fields (
+                    id INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
+                    type_id INTEGER NOT NULL,
+                    parent_type_id INTEGER NOT NULL,
+                    name TEXT NOT NULL
+                )
+
+                CREATE INDEX field_names_by_parent_type ON fields (parent_type_id, name)
+
+                CREATE TABLE field_values (
+                    entity_storage_key_id INTEGER NOT NULL,
+                    field_id INTEGER NOT NULL,
+                    primitive_value_id INTEGER
+                )
+
+                CREATE INDEX field_values_by_entity_storage_key
+                ON field_values (entity_storage_key_id, primitive_value_id)
+
+                CREATE TABLE text_primitive_values (
+                    id INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
+                    value TEXT NOT NULL UNIQUE
+                )
+
+                CREATE INDEX text_primitive_value_index ON text_primitive_values (value)
+
+                CREATE TABLE number_primitive_values (
+                    id INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
+                    value REAL NOT NULL UNIQUE
+                )
+
+                CREATE INDEX number_primitive_value_index ON number_primitive_values (value)
+
+                CREATE TABLE requested_notifiers (
+                    component_package TEXT NOT NULL,
+                    component_class TEXT NOT NULL,
+                    notification_key TEXT NOT NULL
+                )
+
+                CREATE INDEX notifiers_by_component 
+                ON requested_notifiers (
+                    component_package, 
+                    component_class
+                )
```

Comment:
No need for these here..

Reply:
Consider it an homage to my inspiration!